### PR TITLE
Add teraform module

### PR DESCRIPTION
**Pull request by misohu**
Closes: https://github.com/canonical/oidc-gatekeeper-operator/issues/174

Based on this PR: https://github.com/canonical/argo-operators/pull/198

**To test**

- Create juju controller with `kubeflow` model
- Run `terraform apply -var "app_name=oidcgatekeeper" -var "channel=ckf-1.9/stable" -var "model_name=kubeflow" -var "resources={oci-image: 4}"`

### CODE REVIEW

File: terraform/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Terraform module for oidc-gatekeeper
```
Comment (orfeas-k) [inline terraform/README.md:1]: I think we 'd need to include compatibility due to changes in https://github.com/canonical/oidc-gatekeeper-operator/blob/track/ckf-1.9/metadata.yaml (see thread in channel)

File: terraform/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+### Inputs
+The module offers the following configurable inputs:
+
+| Name | Type | Description | Required |
+| - | - | - | - |
+| `app_name`| string | Application name | False |
+| `channel`| string | Channel that the charm is deployed from | False |
+| `config`| map(string) | Map of the charm configuration options | False |
+| `model_name`| string | Name of the model that the charm is deployed on | True |
+| `resources`| map(number) | Map of charm resources revisions | False |
```
Comment (orfeas-k) [inline terraform/README.md:19]: Need to update https://github.com/canonical/mlmd-operator/pull/99/files#r1764719494

File: terraform/variables.tf [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  description = "Model name"
+  type        = string
+}
+
+# TODO: Update to a map of strings, once juju provider 0.14 is released
+variable "resources" {
+  description = "Map of resources revisions"
+  type        = map(number)
+  default     = null
+}
```
Comment (orfeas-k) [inline terraform/variables.tf:29]: Need to be updated https://github.com/canonical/minio-operator/pull/188/files#r1764625697

File: terraform/versions.tf [hunk 4]
```diff
@@ -0,0 +1,9 @@
+terraform {
+  required_version = ">= 1.6"
+  required_providers {
+    juju = {
+      source  = "juju/juju"
+      version = "~> 0.13.0"
+    }
+  }
+}
```
Comment (orfeas-k) [inline terraform/versions.tf:9]: let's update according to canonical/mlmd-operator#99

**Review by orfeas-k — COMMENTED** [review]
Good job, left some comments!

**Review by mvlassis — APPROVED** [review]
Everything looks good, good work!

**Pull request merged**